.github/workflows/update-go-version.yml

```diff
@@ -0,0 +1,90 @@
+          git_commit_gpgsign: true
+
+      - name: Set up Go
+        uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
+        with:
+          go-version: '1.x'
+
+      - name: Get latest Go version
+        id: get-latest-go
+        run: |
+          latest_go_version=$(curl -s https://go.dev/VERSION?m=text)
+          latest_go_version=${latest_go_version#go}  # Strip the 'go' prefix
+          echo "Latest Go version: $latest_go_version"
+          echo "::set-output name=latest_go_version::$latest_go_version"
+
+      - name: Get current Go version from go.mod
+        id: get-current-go
+        run: |
+          current_go_version=$(go mod edit -json | jq -r '.Go')
+          echo "Current Go version: $current_go_version"
+          echo "::set-output name=current_go_version::$current_go_version"
+
+      - name: Check if branch exists on remote
+        id: check-branch
+        run: |
+          branch_name=update-go-version-${{ steps.get-latest-go.outputs.latest_go_version }}
+            if git ls-remote --exit-code --heads origin $branch_name; then
+              echo "Branch $branch_name already exists on the remote."
+              echo "::set-output name=branch_not_exist::false"
+            else
+              echo "::set-output name=branch_not_exist::true"
+            fi
+
+      - name: Check if Go version needs update
+        id: check-update
+        run: |
+            if [ "${{ steps.get-latest-go.outputs.latest_go_version }}" != "${{ steps.get-current-go.outputs.current_go_version }}" ]; then
+              echo "::set-output name=update_needed::true"
+            else
+              echo "::set-output name=update_needed::false"
+            fi
+
+      - name: Update go.mod with latest Go version
+        id: update-go-mod
+        if: ${{ steps.check-update.outputs.update_needed == 'true' && steps.check-branch.outputs.branch_not_exist == 'true'}}
+        run: |
+          latest_go_version=${{ steps.get-latest-go.outputs.latest_go_version }}
+          echo "Updating go.mod to use Go version: $latest_go_version"
+          go mod edit -go=$latest_go_version
+          go mod tidy
+          branch_name=update-go-version-$latest_go_version
+          git checkout -b $branch_name
+          git add go.mod go.sum
+          git commit -S -m "Update Go version to $latest_go_version"
+          git push --set-upstream origin $branch_name
+          echo "::set-output name=branch::$branch_name"
+
+      - name: Install GitHub CLI
+        run: sudo apt-get install gh
+  
+      - name: Create pull request
+        if: ${{ steps.check-update.outputs.update_needed == 'true' && steps.check-branch.outputs.branch_not_exist == 'true'}}
+        run: gh pr create -B main -H ${{ steps.update-go-mod.outputs.branch }} --title "Update Go version to ${{ steps.get-latest-go.outputs.latest_go_version }}" --body "This PR updates the Go version in the go.mod file to the latest available version."
```

Comment:
I think using the `|` syntax and `\`s might be a little more readable(?) For instance:

```
run: |
  gh pr create -B main -H ${{ steps.update-go-mod.outputs.branch }} \
    --title "Update Go version to ${{ steps.get-latest-go.outputs.latest_go_version }}" \
    --body "This PR updates the Go version in the go.mod file to the latest available version."
```

Reply:
I think we should keep it consistent for all of the `run` steps. The other run steps have `|` and don't have `\` but are delineated by a new line. I think a new line is enough to keep it consistent.

Reply:
Isn't there a difference between the earlier command that fits on one line vs. one that is longer and extends over multiple lines? I think the second example needs the `\` to work?

Reply:
Yes I think you're right. I'll add in the `\`